Model/Quote/SetQuoteAddresses.php

```diff
@@ -82,9 +82,7 @@ public function setAddresses(
             $quote->removeAddress($quote->getShippingAddress()->getId());
             $quote->setDataChanges(true);
             $quote->collectTotals();
-            if (!$this->config->isCheckoutTypeSelfHosted((int)$quote->getStore()->getWebsiteId())) {
-                $this->quoteResource->save($quote);
-            }
+            $this->quoteResource->save($quote);
```

Comment:
Do we know why this had an if statement?

Reply:
This if statement was part of some previous functionality that has since been removed for something better. Here's the gist.

In payment booster, we would set the address and the customer within the platform's checkout. Once you navigate to the payment step, the Magento checkout would send a request to bold checkout to set the address and the customer which would then forward the request to the RSA, which then the RSA would tell the platform to set the address and customer even though it already did that on the frontend.

We have since moved away from this and now just send a refresh request from the frontend when we go to the payment step on payment booster.

My best guess as to why this if statement was there to begin with was either this was causing weird order locks on Magento or what's most likely is this was just an optimization to prevent an unnecessary database write.